nautobot/ipam/tests/test_filters.py

```diff
@@ -862,42 +862,46 @@ def setUpTestData(cls):
             VirtualMachine.objects.create(name="Virtual Machine 3", cluster=cluster),
         )
 
-        Service.objects.create(
-            device=devices[0],
-            name="Service 1",
-            protocol=ServiceProtocolChoices.PROTOCOL_TCP,
-            ports=[1001],
-        )
-        Service.objects.create(
-            device=devices[1],
-            name="Service 2",
-            protocol=ServiceProtocolChoices.PROTOCOL_TCP,
-            ports=[1002],
-        )
-        Service.objects.create(
-            device=devices[2],
-            name="Service 3",
-            protocol=ServiceProtocolChoices.PROTOCOL_UDP,
-            ports=[1003],
-        )
-        Service.objects.create(
-            virtual_machine=virtual_machines[0],
-            name="Service 4",
-            protocol=ServiceProtocolChoices.PROTOCOL_TCP,
-            ports=[2001],
-        )
-        Service.objects.create(
-            virtual_machine=virtual_machines[1],
-            name="Service 5",
-            protocol=ServiceProtocolChoices.PROTOCOL_TCP,
-            ports=[2002],
-        )
-        Service.objects.create(
-            virtual_machine=virtual_machines[2],
-            name="Service 6",
-            protocol=ServiceProtocolChoices.PROTOCOL_UDP,
-            ports=[2003],
+        services = (
+            Service.objects.create(
+                device=devices[0],
+                name="Service 1",
+                protocol=ServiceProtocolChoices.PROTOCOL_TCP,
+                ports=[1001],
+            ),
+            Service.objects.create(
+                device=devices[1],
+                name="Service 2",
+                protocol=ServiceProtocolChoices.PROTOCOL_TCP,
+                ports=[1002],
+            ),
+            Service.objects.create(
+                device=devices[2],
+                name="Service 3",
+                protocol=ServiceProtocolChoices.PROTOCOL_UDP,
+                ports=[1003],
+            ),
+            Service.objects.create(
+                virtual_machine=virtual_machines[0],
+                name="Service 4",
+                protocol=ServiceProtocolChoices.PROTOCOL_TCP,
+                ports=[2001],
+            ),
+            Service.objects.create(
+                virtual_machine=virtual_machines[1],
+                name="Service 5",
+                protocol=ServiceProtocolChoices.PROTOCOL_TCP,
+                ports=[2002],
+            ),
+            Service.objects.create(
+                virtual_machine=virtual_machines[2],
+                name="Service 6",
+                protocol=ServiceProtocolChoices.PROTOCOL_UDP,
+                ports=[2003],
+            ),
         )
+        services[0].tags.set(Tag.objects.get_for_model(Service))
```

Comment:
Would it be better if assigning of tags to model is done in the generic test function instead of on each test case

Reply:
You mean have the test function take responsibility for assigning tags, rather than expecting the tags to be provided by the test case setup? The thing is any models that are using factories already have tags appropriately assigned by the factory, the per-test custom setup is an outlier and should be less and less necessary as we implement more factories.